test/e2e/cloud-provider-azure.go

```diff
@@ -26,119 +26,49 @@ import (
-		if err != nil {
-			return err
-		}
-		chartRequested, err := helmLoader.Load(cp)
-		if err != nil {
-			return err
-		}
-		release, err := i.RunWithContext(ctx, chartRequested, vals)
-		if err != nil {
-			return err
-		}
-		Logf(release.Info.Description)
-		return nil
-	}, input.WaitForControlPlaneIntervals...).Should(Succeed())
+	values := []string{fmt.Sprintf("infra.clusterName=%s", input.ConfigCluster.ClusterName)}
+	InstallHelmChart(ctx, input, cloudProviderAzureHelmRepoURL, cloudProviderAzureChartName, cloudProvierAzureHelmReleaseName, values)
 	By("Waiting for a Running cloud-controller-manager pod")
-	Eventually(func() bool {
-		clusterProxy := input.ClusterProxy.GetWorkloadCluster(ctx, input.ConfigCluster.Namespace, input.ConfigCluster.ClusterName)
-		clientSet := clusterProxy.GetClientSet()
-		var runningPods int
-		list, err := clientSet.CoreV1().Pods("kube-system").List(ctx, v1.ListOptions{
-			LabelSelector: "component=cloud-controller-manager",
-		})
-		if err != nil {
-			return false
-		}
-		for _, p := range list.Items {
-			if p.Status.Phase == corev1.PodRunning {
-				runningPods++
-			}
-		}
-		return runningPods > 0
-	}, input.WaitForControlPlaneIntervals...).Should(BeTrue())
+	clusterProxy := input.ClusterProxy.GetWorkloadCluster(ctx, input.ConfigCluster.Namespace, input.ConfigCluster.ClusterName)
+	c := clusterProxy.GetClient()
+	cloudControllerManagerPodLabel, err := labels.Parse("component=cloud-controller-manager")
+	Expect(err).ToNot(HaveOccurred())
+	framework.WaitForPodListCondition(ctx, framework.WaitForPodListConditionInput{
+		Lister: c,
+		ListOptions: &client.ListOptions{
+			LabelSelector: cloudControllerManagerPodLabel,
+			Namespace:     "kube-system",
+		},
+		Condition: podListHasNumPods(1),
+	}, input.WaitForControlPlaneIntervals...)
 	By("Waiting for Running cloud-node-manager pods")
-	Eventually(func() bool {
-		clusterProxy := input.ClusterProxy.GetWorkloadCluster(ctx, input.ConfigCluster.Namespace, input.ConfigCluster.ClusterName)
-		clientSet := clusterProxy.GetClientSet()
-		var runningPods int64
-		list, err := clientSet.CoreV1().Pods("kube-system").List(ctx, v1.ListOptions{
-			LabelSelector: "k8s-app=cloud-node-manager",
-		})
-		if err != nil {
-			return false
-		}
-		for _, p := range list.Items {
-			if p.Status.Phase == corev1.PodRunning {
-				runningPods++
-			}
-		}
-		return runningPods >= to.Int64(input.ConfigCluster.ControlPlaneMachineCount)
-	}, input.WaitForControlPlaneIntervals...).Should(BeTrue())
+	cloudNodeManagerPodLabel, err := labels.Parse("k8s-app=cloud-node-manager")
+	Expect(err).ToNot(HaveOccurred())
+	framework.WaitForPodListCondition(ctx, framework.WaitForPodListConditionInput{
+		Lister: c,
+		ListOptions: &client.ListOptions{
+			LabelSelector: cloudNodeManagerPodLabel,
+			Namespace:     "kube-system",
+		},
+		Condition: podListHasNumPods(int(to.Int64(input.ConfigCluster.ControlPlaneMachineCount) + to.Int64(input.ConfigCluster.WorkerMachineCount))),
```

Comment:
does this also count also need to add windows worker nodes?

Reply:
I think what we're saying is that we need to multiply the `input.ConfigCluster.WorkerMachineCount` by the number of pools in the cluster.

The current OOT template only has a single node pool, which is why this is working.

I don't think we have a way to determine the number of pools from the input to this func, but because we enter into this after all the control plane nodes and machines are online with node references, we should be able to count the number of nodes in the cluster, right?

Reply:
windows is its own variable so we should be able to at least add that to the count (even if it's zero for now, that will make it more future proof)